version: "3.4"

services:
  consume:
    build:
      context: .
      target: local
      dockerfile: consumer/Dockerfile
    depends_on:
      - db
    env_file:
      - .env
    environment:
      POSTGRES_USER: aiven
      POSTGRES_PASSWORD: aiven
      POSTGRES_DB: aiven_default
    networks:
      - default
    command: ["python", "consumer.py"]
    volumes:
      - .:/app

  produce:
    build:
      context: .
      target: local
      dockerfile: producer/Dockerfile
    networks:
      - default
    command: ["python", "producer.py"]
    volumes:
      - .:/app

  db:
    environment:
      POSTGRES_USER: aiven
      POSTGRES_PASSWORD: aiven
      POSTGRES_DB: aiven_default
    image: postgres:11.6-alpine
    networks:
      - default
    ports:
      - 5432:5432
    volumes:
      - ./postgres-data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U aiven -d aiven_default"]
      interval: 10s
      timeout: 5s
      retries: 5